hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/components/navBar/navBar.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import axios, { AxiosResponse } from 'axios';
+import { Layout, Menu, Spin } from 'antd';
+import {
+  BarChartOutlined,
+  ClusterOutlined,
+  ContainerOutlined,
+  DashboardOutlined,
+  DatabaseOutlined,
+  DeploymentUnitOutlined,
+  FolderOpenOutlined,
+  InboxOutlined,
+  LayoutOutlined,
+  PieChartOutlined
+} from '@ant-design/icons';
+import { useLocation, Link } from 'react-router-dom';
+
+
+import logo from '@/logo.png';
+import { showDataFetchError } from '@/utils/common';
+import { AxiosGetHelper, cancelRequests } from '@/utils/axiosRequestHelper';
+
+import './navBar.less';
+
+
+// ------------- Types -------------- //
+type NavBarProps = {
+  collapsed: boolean;
+  onCollapse: (arg0: boolean) => void;
+}
+
+const NavBar: React.FC<NavBarProps> = ({
+  collapsed = false,
+  onCollapse = () => { }
+}) => {
+  const [isHeatmapEnabled, setIsHeatmapEnabled] = useState<boolean>(false);
+  let cancelDisabledFeatureSignal: AbortController;
+  const location = useLocation();
+
+  const fetchDisabledFeatures = async () => {
+    const disabledfeaturesEndpoint = `/api/v1/features/disabledFeatures`;
+    const { request, controller } = AxiosGetHelper(
+      disabledfeaturesEndpoint,
+      cancelDisabledFeatureSignal
+    )
+    cancelDisabledFeatureSignal = controller;
```

Comment:
So the current useEffect() hook makes sure this fetch call is only done once during the initial render.
Hence, in the subsequent re-renders, the `fetchDisabledFeatures()` will not be called again.

So due to this we are sure that no further re-assignments are done to this variable and on unmount it will automatically cancel any pending request, the garbage collector will take care of the variable cleanup.

Reply:
> So the current `useEffect()` hook makes sure this fetch call is only done once during the initial render. Hence, in the subsequent re-renders, the `fetchDisabledFeatures()` will not be called again.
> 
> So due to this we are sure that no further re-assignments are done to this variable and on unmount it will automatically cancel any pending request, the garbage collector will take care of the variable cleanup.

We should correct this someway when user may enable the heatmap flag.

Reply:
So in this scenario, I believe it will be added to the cluster config, and then cluster will be restarted. So user will have to refresh the page anyway

Reply:
> So in this scenario, I believe it will be added to the cluster config, and then cluster will be restarted. So user will have to refresh the page anyway

If user is on Datanodes page, then the heatmap flag is enabled, then user simply refresh that Datanodes page, will HeatMap nav item will display ? or user first has to go to Overview page, then only Heatmap page will display ? Kindly check this part.

Reply:
Yes, if the user is in any page and they refresh, the Heatmap nav item will appear, if it has been enabled
Since refresh means the DOM will completely be unmounted and created new.
Also tested and verified this scenario

Reply:
> Yes, if the user is in any page and they refresh, the Heatmap nav item will appear, if it has been enabled Since refresh means the DOM will completely be unmounted and created new. Also tested and verified this scenario

Ok thanks, I just want to make sure that disabledFeature information on which heatmap hiding and making visible depends on gets refreshed on any page. If this works that way, then we are good to go.